workspaces/ui-v2/src/store/shapes/slice.ts

```diff
@@ -0,0 +1,47 @@
+import { createSlice, SerializedError } from '@reduxjs/toolkit';
+
+import { AsyncStatus } from '<src>/types';
+import { ShapeId, ReduxShape } from './types';
+import { fetchShapes } from './thunks';
+
+const initialState: {
+  rootShapes: Record<ShapeId, AsyncStatus<ShapeId, SerializedError>>;
+  shapeMap: Record<ShapeId, ReduxShape[]>;
+} = {
+  rootShapes: {},
+  shapeMap: {},
+};
+
+const shapesSlice = createSlice({
+  name: 'shapes',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    // TODO we probably need to refetch shapes if they get outdated
```

Comment:
When we come to actually make changes to shapes, we'll need to figure out if we need to refetch shapes (which means we should reset all shapes or a subset if we can be sure only some shapes changed).